src/client.js

```diff
@@ -1740,16 +1740,51 @@ class Client {
     async getSchema(schemaId, representation, internal) {
         var entity = await this._entityCache.get("xtk:schema", schemaId);
         if (!entity) {
-          entity = await this.getEntityIfMoreRecent("xtk:schema", schemaId, "xml", internal);
-          if (entity) {
-            const impls = DomUtil.getAttributeAsString(entity, "implements");
-            if (impls === "xtk:persist" && schemaId !== "xtk:session" && schemaId !== "xtk:persist") {
-                // Ensure xtk:persist is present by loading the xtk:session schema
-                await this.getSchema("xtk:session", "xml", true);
+              // special case of "temp:group:*" schemas for nms:group
+              // Schema "temp:group:*" is not cached because life cycle of this kind of schema is not the same as the others schemas
+              if (schemaId.startsWith("temp:group:")) {
+                  const parts = schemaId.split(":");
+                  var queryDef = {
+                    "schema": "nms:group",
+                    "operation": "get",
+                    "select": {
+                      "node": [
+                        { "expr": "@id" },
+                        { "expr": "extension" }
+                      ]
+                    },
+                    "where": {
+                      "condition": [
+                        { "expr": "@id=" + XtkCaster.asLong(parts[2]) }
+                      ]
+                    }
+                  };
+                  // Convert to current representation
+                  queryDef = this._convertToRepresentation(queryDef, "SimpleJson", "xml");
+                  const query = this.NLWS.xml.xtkQueryDef.create(queryDef);
+                  try {
+                      const groupSchema = await query.executeQuery(internal);
+                      const extension = DomUtil.findElement(groupSchema, "extension");
+                      if (extension) {
+                          entity = extension;
+                      } else {
+                          entity = null;
```

Comment:
You should test the error code and return null if the group or schema is not found only. There are all kind of exceptions, such as network issues, token expiration, etc. and we want those to be reported. In case the schema is not found, we return null because that's the contract of the function

Reply:
I have set entity to null only for CampaignExceptions ,for others exceptions I re-throw the exception.

Reply:
No, you have to set entity to null if the group does not exist of if does not have an extension node. CampaignException can represent tons of errors. You have to check actual error code return carefully.

Also, do not forget to have the corresponding unit tests: execption cases needs to be tested too